suse2021-ns/common/navigation.xsl

```diff
@@ -111,14 +111,13 @@
   <!-- ===================================================== -->
   <xsl:template name="is.xref.in.samebook">
     <xsl:param name="target" select="NOT_A_NODE"/>
-
-    <xsl:variable name="target.book" select="($target/ancestor-or-self::d:article|$target/ancestor-or-self::d:book)[1]"/>
-    <xsl:variable name="this.book" select="(ancestor-or-self::d:article|ancestor-or-self::d:book)[1]"/>
-
-    <xsl:choose>
-      <xsl:when test="(generate-id($target.book) = generate-id($this.book))">1</xsl:when>
-      <xsl:otherwise>0</xsl:otherwise>
-    </xsl:choose>
+    <xsl:variable name="context" select="."/>
+    <xsl:variable name="this.ancestor" select="ancestor-or-self::*[@xml:id=$rootid]"/>
+    <xsl:variable name="target.or.none" select="$this.ancestor//*[@xml:id=$context/@linkend]"/>
+    <!-- Alternative, but not similar way:
+         $target/ancestor-or-self::*[@xml:id=$rootid] = $this.ancestor
+    -->
+    <xsl:value-of select="count($target.or.none)"/>
```

Comment:
Maybe. It's true, the code doesn't take that into account. This can only happen, if we have multiple IDs with the same value. However, I assume, this is handled outside with profiling, (transclusion?) and validation.

Reply:
You are of course right. And I think your checks use `!= 0` anyway. -> Resolving.